### DISCUSSION TITLE
AWS Aurora Support

### DISCUSSION THREAD

**Discussion by timkendall** [readysettech/readyset#28] [Q&A] [answered]
In my (limited) tests, the currently published distribution (Docker image) runs agains an AWS RDS Postgres instance but not an AWS Aurora Postgres cluster. The following error is logged when running against aurora:

> replicators: Unrecoverable error in replication, restarting after restart timeout error=Error during replication: PostgreSQL: db error: ERROR: must be superuser or replication role to run this operation.

AWS Aurora [claims to support logical replication ](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.Replication.Logical.html) however similar projects utilizing logical replication seem to potentially have the same issue (https://github.com/supabase/realtime/issues/270, [Debezium Google groups thread](https://groups.google.com/g/debezium/c/YHotnMwBIDk?pli=1)).

Has anyone gotten Readyset working with Aurora?

**Comment by jbensal** [comment]
Hi @timkendall,

Thanks so much for your report. Let me check with the team - I don't think we've officially tested Aurora support yet. However, I'd love to chase this down with you and see if we can get ReadySet working with your Aurora Postgres cluster.

**Reply by timkendall** [reply]
Hey @jbensal I appreciate that. I've done some more digging and here is what I found.

- Current Docker tag works with Aurora Postgres 14.5 ("works" as in I connected it and cached one query ;)
- Current main works with Aurora Postgres 13.7 (where I was getting the aforementioned error) **if** `SHOW server_version_num` is not executed in https://github.com/readysettech/readyset/blob/main/replicators/src/postgres_connector/connector.rs#L264 

No clue what's up with that, maybe an Aurora bug. I'm sure there's probably some other small things to discover like that. 

Is this the place to raise these questions/report findings? Awesome project. Hope you guys make it :)

**Reply by jbensal** [reply]
Thank you for digging more - that's super helpful. Is there a particular reason you're interested in using Aurora over RDS?

Either here or our Discord community is the right spot. We'll see it in both cases.

Thanks for the kind words! We hope so too 🤞

**Reply by timkendall** [reply]
My current company switched from RDS to Aurora a while back. I believe the reasoning was to increase our IOPS headroom on an already maximally (vertically) scaled RDS instance.

And noted 👍

**Comment by lukoktonos** [comment, marked as answer]
Hi @timkendall, thanks for starting the discussion here and even doing the initial debugging work to pinpoint the issue with `SHOW server_version_num`!

We determined the root cause of this was that for PostgreSQL versions below 14, we run some SQL commands that turn the connection into a "replication connection", which Aurora has configured to not allow `SHOW` statements. [ddl_replication.sql](https://github.com/readysettech/readyset/blob/main/replicators/src/postgres_connector/ddl_replication.sql).

I moved this query to before the ddl setup [in this commit](https://github.com/readysettech/readyset/commit/9be644320a7521d8f2c30c3fda811c7736c4fb26), which allows ReadySet to spin up on RDS Aurora using Postgres 13.

**Reply by timkendall** [reply]
Nice! 👍🏼
